EIPS/eip-3224.md

```diff
@@ -0,0 +1,465 @@
+can provide meaningful feedback to the user describing the
+action the user is about to perform.
+
+
+## Motivation
+
+When using an Ethereum Wallet (e.g. MetaMask, Clef, Hardware
+Wallets) users must accept and authorize signing messages or
+sending transactions.
+
+Due to the complexity of Ethereum transactions, wallets are very
+limitd in their ability to provide insight into the contents of
+transactions user are approving; outside special-cased support
+for common transactions such as ERC20 transfers, this often amounts
+to asking the user to sign an opaque blob of binary data.
+
+This EIP presents a method for dapp developers to enable a more
+comfortable user experience by providing wallets with a means
+to generate a better description about what the contract claims
+will happen.
+
+It does not address malicious contracts which wish to lie, it
+only addresses honest contracts that want to make their user's
+life better. We believe that this is a reasonable security model,
+as transaction descriptions can be audited at the same time as
+contract code, allowing auditors and code reviewers to check that
+transaction descriptions are accurate as part of their review.
+
+
+## Specification
+
+The **description string** and **described data** are generated
+simultaneously by evaluating **describer** bytecode, passed
+**describedInput** as the calldata with the encoded method:
+
+```solidity
+function eipXXXDescribe(bytes inputs) view returns (string description, bytes data);
+```
+
+The method MUST be executable in a static context, i.e. any
+side effects (logX, sstore, etc., (including through indirect
+calls) will be ignored.
+
+During evaluation, the `ADDRESS` (i.e. `to`), `CALLER`
+(i.e. `from`), `VALUE`, and `GASPRICE` must be the same as the
+values for the transaction being described, so that the
+code generating the description can rely on them. For signing
+**described messages**, `VALUE` should always be 0.
+
+When executing the bytecode, best efforts should be made to
+ensure `BLOCKHASH`, `NUMBER`, `TIMESTAMP` and `DIFFICULTY`
+match the `"latest"` block. The `COINBASE` should be the zero
+address.
+
+If evaluating the function results in a revert, and the return
+data is of a length that is congruent to `4 mod 32`, and the
+first 4 bytes represent the selector `Error(string)`, the
+reason string should be displayed to the user. If evaluation
+reverts otherwise, then a generic evaluation error should be shown.
+In either case, the signing is aborted.
+
+
+### New JSON-RPC Methods
+
+Clients which manage private keys should expose additional
+methods for interacting with the related accounts.
+
+If an user interface is not present or expected for any other
+account-based operations, the description strings should be
+ignored and the described data used directly.
+
+These JSON-RPC methods will also be implemented in standard
+Ethereum libraries, so the JSON-RPC description is meant more
+of a canonical way to describe them.
```

Comment:
Above you are specifying a contract interface (ERC) and here you are specifying new JSON-RPC methods (Interface).  This should be split into two EIPs, where perhaps the JSON-RPC one has a dependency on the contract interface one.

Reply:
I will bring up splitting this EIP up on the discussion-to link, to help figure out exactly where the lines should be drawn. :)